byway/src/main/webapp/routepage.js

```diff
@@ -292,5 +292,45 @@ function createRecButton(rec){
   return recBtn;
 }
 
-/* exported initMap, generateRoute, placesService, map */
-/* global google, findPlace, getTripKeyFromUrl, configureTripKeyForPath */
\ No newline at end of file
+
+/** Creates a URL link to Google Maps based on the start/end point and route
+ *  @returns {String} routeLink url containing query params for the user’s route
+ */
+function generateRouteLink(){
+    let routeLink = "https://www.google.com/maps/dir/?api=1&travelmode=driving" 
+    routeLink += "&origin=" + start.name.split(' ').join('%20') + "&destination=" + 
+      end.name.split(' ').join('%20') + "&waypoints=";
+    for(let place of route){
+      routeLink += place.name.split(' ').join('%20') + "|";
+    }
+   return routeLink;
+}
+
+/** Determines if email address is valid using regex patterns
+ *  @param {String} email email address 
+ *  @returns {boolean} is the email a valid email address
+ */
+function validateEmail(email) {
+    const re = /^\w+([-]?\w+)*@\w+([-]?\w+)*(\.\w{2,3})+$/;
```

Comment:
Email is regex is overly strict, bans longer domain names like `.info`, and doesn't support emails with `+` and other chars in the name.

You could look at https://www.regular-expressions.info/email.html for a regex that matches email addresses more generally.

Reply:
Alright, I used a regex pattern listed on that site which supports 99% of email addresses.